src/components/Deployments/columns.tsx

```diff
@@ -0,0 +1,275 @@
+  };
+
+  catalogEnvironment(env: string): string {
+    const index = this.catalogEnvironmentsLower.indexOf(env.toLowerCase());
+    if (index >= 0) {
+      return this.catalogEnvironments[index];
+    }
+    return env;
+  }
+
+  catalogEnvironmentSort(envA: string, envB: string): number {
+    let sortVal = 0;
+    const aLower = this.catalogEnvironment(envA).toLowerCase();
+    const bLower = this.catalogEnvironment(envB).toLowerCase();
+
+    if (this.catalogEnvironments.length == 0) {
+      sortVal = aLower < bLower ? -1 : 1;
+    } else {
+      if (aLower !== bLower) {
+        for (let i in this.catalogEnvironments) {
+          const envLower = this.catalogEnvironments[i].toLowerCase();
+          if (aLower == envLower) {
+            sortVal = 1;
+            break;
+          }
+          if (bLower == envLower) {
+            sortVal = -1;
+            break;
+          }
+        }
+      }
+    }
+    return sortVal;
+  }
+
+  createIdColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'ID',
+      field: 'id',
+      highlight: false,
+      hidden: true,
+    };
+  }
+
+  createLatestColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'Latest',
+      field: 'Latest',
+      highlight: false,
+      width: '100px',
+      render: (row: EnvDeployment) => (row.latest ? <CheckIcon /> : undefined),
+      lookup: {
+        true: '✔',
+        false: undefined,
+      },
+    };
+  }
+
+  createEnvironmentColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'Environment',
+      field: 'environment',
+      highlight: false,
+      width: '100px',
+      customSort: (a: EnvDeployment, b: EnvDeployment) =>
+        this.catalogEnvironmentSort(a.environment, b.environment),
+      render: (row: EnvDeployment) => {
+        const env = this.catalogEnvironment(row.environment);
+        const context = this.getContext(row.payload.workflow_run_url as string);
+        const url = `https://${context.server}/${context.owner}/${context.repo}/deployments/${env}`;
+        return this.buildLink(url, env);
+      },
+    };
+  }
+
+  tagRegExp: RegExp = /^(v?\d+(?:\.\d+)*.*)$/g;
+  shaRegExp: RegExp = /\b([0-9a-f]{40})\b/g;
+
+  createTagRefColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'Branch/Tag/SHA',
+      field: 'ref',
+      highlight: false,
+      customSort: (a: EnvDeployment, b: EnvDeployment) => {
+        const aVersion = this.parseVersionString(a.ref);
+        const bVersion = this.parseVersionString(b.ref);
+        const versionDiff =
+          aVersion.major - bVersion.major ||
+          aVersion.minor - bVersion.minor ||
+          aVersion.build - bVersion.build;
+
+        if (versionDiff == 0) {
+          if (aVersion.version == bVersion.version) {
+            return 0;
+          } else {
+            return aVersion.version < bVersion.version ? -1 : 1;
+          }
+        } else {
+          return versionDiff;
+        }
+      },
```

Comment:
You added `compare-versions` to package.json, but you've implemented your own semver parsing and comparison here. Which one do you want to keep? And if you don't want to or can't use compare-versions, why not use at least the `semver` package to detect if the ref is a semver and parse it if so?

Also, semver has rules on version comparisons that go beyond just the major, minor, and patch, and also include the pre-release label's segments.